Proll 276/main page design update

Fermatka opened the pull request:
- [x] Update main page design, according to FIGMA
- [x] Check responsiveness of main page
- [x] Check main page darkMode if needed
- [x] Create Heavy industry Page
- [x] Create Med- Tech page
- [x] Check everything together
- [x] Add color highlighting to the selected option in the navbar: [little task link](https://sacrebleu.atlassian.net/browse/PROLL-174l)

[TASK LINK](https://sacrebleu.atlassian.net/browse/PROLL-276)
Work in progress:)

Fermatka committed:
DataLabSection added

Fermatka committed:
new verion added to RoundedTextComponent

Fermatka committed:
basic OurWork design done

Fermatka committed:
DataLabSection and OurWork updated with correct styles

Fermatka committed:
SoftwareConsultingSection created

Fermatka committed:
OurWork componed updated with proper styles

Fermatka committed:
HeavyIndustry and MedTech pages added with mock layout, added new routes, added link handling in OurWork section

Fermatka committed:
DescriptionComponentTexts added

Fermatka committed:
DescriptionComponent done with proper text versions, need to update image

Fermatka committed:
font sizes updated in DescriptionImageComponent

Fermatka committed:
CollapsibleSectionsComponent added, need to refactor some logic still

Fermatka committed:
Upadated OfferSection, need to refactor handleOffer function and add proper texts in OfferSectionTexts

Fermatka committed:
Image added to OpportiunitiesIndustrySection

Fermatka committed:
Image added to OpportunitiesSection

Fermatka committed:
Added rendering of the image depending on the version in DescriptionImageComponent

Fermatka committed:
background images added to OurWork.js

Fermatka committed:
OurWork bg images updated with proper styles

Fermatka committed:
updated styles to correct Dark-LightMode changing

Fermatka committed:
images corrected

Fermatka committed:
Basic design of DiagramComponent added, new Icon Components created

Fermatka committed:
DiagramComponent done with basic styles and responsiveness

Fermatka committed:
HeavyIndustry  and medTech pages updated

Fermatka committed:
Diagram Component updated with correct styles

Fermatka committed:
OurWork upaded with proper styles

Fermatka committed:
styles checking with FIGMA

Fermatka committed:
corrected CollapsibleSectionsComponent error, corrected problem with images sizes in HeavyIndustry and Medtech pages

Fermatka committed:
responsiveness corrections

Fermatka commented:
I've changed one component, as I think it's better visible, but please write me what you think:
my solution:
![Algrtm _ Algorithmic solutions - Google Chrome 08 05 2024 10_28_20](https://github.com/Nospoko/algrtm/assets/118021948/9fffc6e2-f303-4508-90b3-bfc65335d93e)
FIGMA:
![Algrtm (Copy) – Figma - Google Chrome 08 05 2024 10_27_33](https://github.com/Nospoko/algrtm/assets/118021948/6d1909b9-5ed2-4c04-b8a3-457d7926c24a)

TheMankel commented on algrtm/src/components/DataLabSection.js:
```diff
@@ -0,0 +1,27 @@
+
+const DataLabSection = () => {
+  return (
+    <section className="w-full flex flex-col gap-6 justify-center items-center text-center mb-[2.75rem] lg:my-[5.88rem] mt-[2.75rem] mx-[1rem] md:max-w-[54.875rem] md:mt-[7.75rem] md:mx-[4.56rem] xl:mx-[5.88rem] bg-neutralLight-neutral100 dark:bg-neutralDark-neutral100">
+      <h4 className="h4 lg:h2SB xl:h2 text-start text-neutralLight-neutral10 dark:text-neutralDark-neutral10">
+        Our Data Laboratory
+      </h4>
+      <div className="flex flex-col gap-[2rem] lg:flex-row w-full self-stretch">
+        {dataLabSectionTexts.map((item, index) => (
+          <div
+            key={index}
+            className="flex flex-col text-start bg-neutralLight-neutral90 border-neutralLight-neutral90  border-1 rounded-[2rem] gap-[1rem] p-[2rem] w-full max-w-[24.75rem]"
+          >
+            <h5 className="h6_600 lg:text-[1.625] lg:leading-[2.34rem] lg-text-semibold">
+              {item.title}
+            </h5>
+            <p className="p2">{item.description}</p>
+          </div>
+        ))}
+      </div>
+    </section>
```
![image](https://github.com/Nospoko/algrtm/assets/26410548/42ca9e5d-505c-4caa-aae4-d30cac28bbf2)

These columns are squeezed because of `md:max-w-[54.875rem]`. After removing it and adding `justify-center` to inner div it should use full width

![image](https://github.com/Nospoko/algrtm/assets/26410548/7f077d50-cabd-441d-bbcf-0956efc55a78)

TheMankel commented on algrtm/src/components/DevelopSection.js:
```diff
@@ -5,7 +5,7 @@ const DevelopSection = () => {
     <section className="flex flex-col gap-6 justify-center items-center text-center max-w-[21.4375rem] mt-[2.75rem] mx-[1rem] md:max-w-[54.875rem] md:mt-[7.75rem md:mx-[4.56rem] xl-mx-[17.56rem] bg-neutralLight-neutral100 dark:bg-neutralDark-neutral100">
       <div className="h2 md:h1 text-neutralLight-neutral10 dark:text-neutralDark-neutral10 leading-[6.02rem]">
-        We develop algorithmic solutions
+        Make Data Science Great Again
```
Probably this text should be wrapped by h2 element and not by div to make it more semantic. The section should also have the margin set to 0 on small devices, because the parent div uses padding (so now with the margin it has 1rem added twice)

![image](https://github.com/Nospoko/algrtm/assets/26410548/84984018-dc15-4144-80c0-de8e8b3e3eab)

TheMankel commented on algrtm/src/components/OpportunitiesIndustrySection.js:
```diff
@@ -0,0 +1,36 @@
+      </h2>
+      <div className="flex flex-col gap-4 w-full xl:px-[5.875rem]">
+        <div className="flex flex-col md:flex-row gap-[1.5rem]  bg-neutralLight-neutral90 dark:bg-neutralDark-neutral90 px-[2.75rem] py-[2.75rem] md:py-[5.88rem]  xl:py-[5.85rem] xl:px-[6.75rem] rounded-[2rem]">
+          <div className=" h5_small_bold md:h4SB lg:h4 text-neutralLight-neutral10 dark:text-neutralDark-neutral10 flex justify-start max-w-[25.1875rem] w-full">
+            <Image
+              src="/factory-image.png"
+              alt="Image of the factory inside"
+              width={440}
+              height={554}
+              sizes="100vw"
+              style={{
+                width: "100%",
+                height: "auto",
+              }}
+            />
+          </div>
+
+          <CollapsibleSectionsComponent version="version3" />
+        </div>
```
Something weird is happening here with the image. When we reveal the text, the image stretches:

![image](https://github.com/Nospoko/algrtm/assets/26410548/6efa9f68-a320-4c91-b116-b285dc5dc589)

On smaller devices, the text should not be centered, and there is something wrong with the icons:

![image](https://github.com/Nospoko/algrtm/assets/26410548/64c6155f-258a-4e45-8c59-f7d1a3f38dea)

TheMankel commented on algrtm/src/components/OurWork.js:
```diff
@@ -0,0 +1,73 @@
+        Our work
+      </h4>
+      <div className="flex flex-col items-center justify-center w-full lg:flex-row gap-[1.5rem] md:flex-row xl:gap-[2rem] mt-[2rem]">
+        <div className=" flex flex-col justify-end items-center self-stretch pt-[9.29169rem] pb-[0.02081rem] rounded-[2rem] w-[21.44rem] text-neutralLight-neutral100 bg-[url('/heavyIndustry-bg-image.jpeg')]">
```
This div and others with `bg-[some_url]` have shifted images and different sizes than on Figma. Adding such styling should fix this: `w-full md:w-[20.25rem] h-[25rem] bg-center bg-cover`

TheMankel reviewed (commented):
Mostly looks very good, I just added some minor comments!

As for that component you changed - I like your version more, because in the original one (Figma) the text fragment mixes with different background colors and in my opinion is less readable 👌

roszcz commented:
Some issues with text alignment:

<img width="441" alt="image" src="https://github.com/Nospoko/algrtm/assets/8056825/ead53672-5268-4d76-bb70-e52e35c61ed8">

HolaAleksandra commented:
I noticed that the Our Data Laboratory section looks different in design and development. I'm not sure if these changes are intentional, but in terms of design, "Our Data Laboratory" is left-aligned (now it's centered), and the texts look different too - in design, they always start on the same line height, and subtitles have different divisions. :)
<img width="1439" alt="Screenshot 2024-05-09 at 10 13 02" src="https://github.com/Nospoko/algrtm/assets/120928465/eca8d4c3-a8ea-498b-a85b-29e83b65f31b">
<img width="1432" alt="Screenshot 2024-05-09 at 10 12 50" src="https://github.com/Nospoko/algrtm/assets/120928465/17506bd1-3adf-4317-8baa-b4eba99feae0">

HolaAleksandra commented:
I found one more small thing - the buttons on Our Work section in development are different than in the design :)
<img width="1188" alt="Screenshot 2024-05-09 at 10 30 12" src="https://github.com/Nospoko/algrtm/assets/120928465/0b798f2c-85e2-4f6e-9186-24ea6e383662">
<img width="767" alt="Screenshot 2024-05-09 at 10 30 21" src="https://github.com/Nospoko/algrtm/assets/120928465/854000f3-a516-4351-a1f0-17dd5542683e">

Fermatka committed:
DataLabSection corrected

Fermatka committed:
Buttons in OurWork corrected

Fermatka committed:
OpportunitiesIndustrySection text aligment and image height corrected

Fermatka committed:
CollapsibleSectionComponent updated with correct styles

Fermatka committed:
Ourwork section updated with correct bg images sizes

Fermatka committed:
Added color highlighting to the selected option in the navbar

TheMankel commented on algrtm/src/components/DataLabSection.js:
```diff
@@ -0,0 +1,27 @@
+const DataLabSection = () => {
+  return (
+    <section className="w-full flex flex-col gap-6 mb-[2.75rem] lg:my-[5.88rem] mt-[2.75rem] mx-[1rem] md:mt-[7.75rem] md:mx-[4.56rem] xl:mx-[5.88rem] bg-neutralLight-neutral100 dark:bg-neutralDark-neutral100">
+      <div className="h4 lg:h2SB xl:h2 mx-[1rem] md:mx-[4.56rem] xl:mx-[5.88rem] text-neutralLight-neutral10 dark:text-neutralDark-neutral10">
+        Our Data Laboratory
+      </div>
```
This text moves to the left, while the cards remain centered, so it looks weird on higher resolutions

![image](https://github.com/Nospoko/algrtm/assets/26410548/cc6eea78-b85e-482c-b63b-855d09c5a666)

We can try to limit the width of the div with the text to be aligned with the first card

Fermatka replied:
I suggest to center the title too, looks much better to me, write me what you think:)

TheMankel replied:
Honestly, both look good to me 😄

TheMankel reviewed (commented):
The image in `/heavyIndustry` has white corners while dark theme is applied:

![image](https://github.com/Nospoko/algrtm/assets/26410548/49bb5b54-2cef-479b-ac78-3af159d476e9)


Three icons doesn't scale down on lower resolutions:

![image](https://github.com/Nospoko/algrtm/assets/26410548/905b80f3-0239-4720-a876-79a68afc76b2)

We also have some errors in the console, and they seem to occur only when dark mode is used:

![image](https://github.com/Nospoko/algrtm/assets/26410548/c2ef53a8-a0cf-4e0f-8cf5-31d88f0c78bf)

But I think we can leave these dark mode problems for a separate task, as we discussed recently

Fermatka committed:
changed main page components order changes for tests

Fermatka committed:
changed place of chatbots info

TheMankel commented:
Just an idea, but when I look at the site now, as a potential customer/user after the current reordering of components, I miss the information at the very top about who we really are and what we do. Hence the idea that maybe this layout would be slightly better:

![image](https://github.com/Nospoko/algrtm/assets/26410548/7f49900d-93f9-4315-8cfa-2929f223eecd)

Let me know what you think 🤯

Fermatka committed:
changed image in OpportunitiesIndustrySection, need to add proper size

Fermatka committed:
TechIconsComponent updated and factory image updated with correct size settings

Fermatka committed:
AWS.js updated

Fermatka committed:
Ourwork.js modified, added images brightness to correct texts visuality

Fermatka commented:
I modified images in OurWork,js in FIGMA it's said there should be 80% of brightness, but I've added 50% as for me it's better visible, please tell we what you think:

![Zrzut ekranu 2024-05-13 123704](https://github.com/Nospoko/algrtm/assets/118021948/ac091e55-583f-4a1a-8920-54962d87a470)

TheMankel reviewed (approved):
> I modified images in OurWork,js in FIGMA it's said there should be 80% of brightness, but I've added 50% as for me it's better visible, please tell we what you think:

I think it is much more visible after this change 🚀
